dtos/devicechangednotification.go

```diff
@@ -1,26 +1,31 @@
-// Copyright (C) 2022 IOTech Ltd
+// Copyright (C) 2022-2023 IOTech Ltd
 
 package dtos
 
 import (
 	"encoding/json"
+	"time"
 )
 
 type DeviceNotificationContent struct {
-	DeviceName        string `json:"deviceName"`
-	DeviceServiceName string `json:"deviceServiceName"`
-	ActionType        string `json:"actionType"`
-	OperatingState    string `json:"operatingState"`
-	AdminState        string `json:"adminState"`
+	DeviceName        string                        `json:"deviceName"`
+	DeviceServiceName string                        `json:"deviceServiceName"`
+	Protocols         map[string]ProtocolProperties `json:"protocols"`
+	ActionType        string                        `json:"actionType"`
+	OperatingState    string                        `json:"operatingState"`
+	AdminState        string                        `json:"adminState"`
+	Datetime          string                        `json:"datetime"`
 }
 
 func NewDeviceNotificationContent(device Device, action string) DeviceNotificationContent {
 	return DeviceNotificationContent{
 		DeviceName:        device.Name,
 		DeviceServiceName: device.ServiceName,
+		Protocols:         device.Protocols,
 		ActionType:        action,
 		OperatingState:    device.OperatingState,
 		AdminState:        device.AdminState,
+		Datetime:          time.Now().Format(time.RFC1123),
```

Comment:
Use Timestamp (int64) instead of DateTime, because it's easy to convert timestamp to RFC1123 if users want

Reply:
Notifications model contains DB Timestamp, so it is fine to use Datetime here.